lib/proxy/helpers.rb

```diff
@@ -25,7 +25,8 @@ def log_halt code=nil, exception=nil
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    ssl_client_cert_header = Proxy::SETTINGS.ssl_client_cert_header || 'SSL_CLIENT_CERT'
+    certificate_raw = request.env[ssl_client_cert_header].to_s.gsub(/   /, "\n")
```

Comment:
I'd prefer not to make this fix specific to nginx (or this version of it) and handle the issue in accordance with rfc:

```
def strip_header_continuation_from_ssl_cert(cert_header_value)
  begin_cert = "-----BEGIN CERTIFICATE-----"
  end_cert = "-----END CERTIFICATE-----"
  cert_body_with_header_continuation = cert_header_value.slice((begin_cert.length)..-(end_cert.length+1))
  begin_cert + "\n" + cert_body_with_header_continuation.gsub(/\s/, "\n") + "\n" + end_cert
end
``` 

I think the new line before "END CERTIFICATE" is needed, could you please check that? Also, please note that the regex in gsub now matches single white spaces (per rfc).